app/controllers/application_controller.rb

```diff
@@ -1544,10 +1544,11 @@ def prov_redirect(typ = nil)
     @org_controller = "vm" # request originated from controller
     @refresh_partial = typ ? "prov_edit" : "pre_prov"
     if typ
-      @prov_id = find_record_with_rbac(VmOrTemplate, checked_or_params).id
+      prov_obj = find_record_with_rbac(VmOrTemplate, checked_or_params)
```

Comment:
I don't understand the implications here, but did you considered `s/prov_obj/@record/` ?

Reply:
Yes I did/still am but I'm not sure that every case that hits here is covered by vm_clone, vm_migate, or vm_publish